### PR TITLE
perf: refresh add plugin treeview item

### PR DESCRIPTION
**Pull request by yuqizhou77**
- refresh isDeclarativeCopilot to update development tree view 5 seconds after user saving manifest.json

### CODE REVIEW

File: packages/vscode-extension/src/manifestListener.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  AppPackageFolderName,
+  ManifestTemplateFileName,
+  TeamsAppManifest,
+} from "@microsoft/teamsfx-api";
+import TreeViewManagerInstance from "./treeview/treeViewManager";
+import { ExtTelemetry } from "./telemetry/extTelemetry";
+import { TelemetryEvent, TelemetryProperty } from "./telemetry/extTelemetryEvents";
+import { isValidProjectV3 } from "@microsoft/teamsfx-core";
+
+function setAbortableTimeout(ms: number, signal: any) {
```
Comment (tecton) [inline packages/vscode-extension/src/manifestListener.ts:23]: You could take a look at [debounce](https://www.geeksforgeeks.org/lodash-_-debounce-method/) function.

File: packages/vscode-extension/src/manifestListener.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      // Clear the timeout and reject the promise if aborted
+      clearTimeout(timeoutId);
+      reject("resolved after clear");
+    });
+  });
+}
+
+export function manifestListener(): vscode.Disposable {
+  let abortController: undefined | AbortController;
+  const disposable = vscode.workspace.onDidSaveTextDocument(
```
Comment (tecton) [inline packages/vscode-extension/src/manifestListener.ts:41]: AFAIK, this triggers when the manifest file is edited in VS Code. If you want to monitor file changes outside VS Code, you could use `createFileSystemWatcher` to monitor change events.

**Review by tecton — APPROVED** [review]

File: packages/vscode-extension/src/manifestListener.ts (hunk 2 above)
Comment (yuqizhou77) [reply packages/vscode-extension/src/manifestListener.ts:41]: Yes, we only need to track when user edit the file in vscode.

File: packages/vscode-extension/src/manifestListener.ts (hunk 1 above)
Comment (yuqizhou77) [reply packages/vscode-extension/src/manifestListener.ts:23]: Thanks for letting me know the alternative :)

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by anchenyi — APPROVED** [review]

**Pull request merged**